components/style/values/computed/border.rs

```diff
@@ -81,3 +82,30 @@ impl ToAnimatedZero for BorderCornerRadius {
         Err(())
     }
 }
+
+/// The computed value of the `border-image-repeat` property:
+///
+/// https://drafts.csswg.org/css-backgrounds/#propdef-border-image-repeat
+#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToCss)]
+pub struct BorderImageRepeat(pub RepeatKeyword, pub RepeatKeyword);
+
+impl BorderImageRepeat {
+    /// Returns the `stretch stretch` value.
+    pub fn repeat() -> Self {
+        BorderImageRepeat(RepeatKeyword::Stretch, RepeatKeyword::Stretch)
+    }
+}
+
+impl ToComputedValue for SpecifiedBorderImageRepeat {
+    type ComputedValue = BorderImageRepeat;
+
+    #[inline]
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        BorderImageRepeat(self.0, self.1.unwrap_or(self.0))
+    }
+
+    #[inline]
+    fn from_computed_value(computed: &Self::ComputedValue) -> Self {
+        SpecifiedBorderImageRepeat(computed.0, Some(computed.1))
+    }
+}
```

Comment:
I think you can `impl ToComputedValue for BorderImageRepeat` inside `specified` module like `BorderSideWidth` in `specified/border.rs`.

Reply:
Oh yeah, it should have been in specified/border.rs. Thanks for catching that :)

Reply:
Although, come to think of it, the reason BorderSideWidth can be in specified/border.rs is that its computed value is particularly simple (computed::NonNegativeLength). I don't think makind specified/border.rs depend on computed/border.rs (for its ComputedValue) would be a good idea.

Reply:
I think it's okay to make it depend on `computed/border.rs`.

You can also check `specified/font.rs`.
It import `font as computed`

https://github.com/servo/servo/blob/657d8a2ede97e7f7fedee5ada97c833abe5a8d9e/components/style/values/specified/font.rs#L22

and use in the specified module.

Ex.
https://github.com/servo/servo/blob/657d8a2ede97e7f7fedee5ada97c833abe5a8d9e/components/style/values/specified/font.rs#L1122-L1142

Reply:
I see, that is good to know! 

I don't mean to be difficult, but I wanted to understand why one way is better than the other, as they look mostly the same to me. I grepped for "impl ToComputedValue" and specified/ has 33 impls (12 of them in font.rs), while computed/ has 15 impls (including background-repeat, on which I based this). I understand you have been working on this a lot, so a completely valid reason would be something like "Consistency is good and I prefer us to standardise on this" :)

Reply:
We're not consistent at this at all, I filed https://github.com/servo/servo/issues/19670. IIRC @nox had a stronger opinion than me at this regarding dependencies from `computed` to `specified` and vice-versa, so if it's not very urgent we could wait for different arguments to come in there and then make a decision and move all of them :)